### PR TITLE
Add more ops: Sigmoid, (Two)MatVecAdd. Faster TwoMatVec.

### PR DESCRIPTION
**Pull request by veluca93**
drive-by: some build system simplifications

### CODE REVIEW

File: ops.h [hunk 1]
```diff
@@ -191,15 +238,15 @@ HWY_INLINE void FullDotProductsForStrip(
...
-// Stores dot products of rows with `vec_aligned` to a buffer, then stores them
-// to `out`.
-template <size_t kOuter, size_t kInner, typename MatT, size_t kCapacity,
-          typename VecT>
-HWY_INLINE void MatVec(const CompressedArray<MatT, kCapacity>& mat,
-                       const size_t mat_ofs,
-                       const VecT* HWY_RESTRICT const vec_aligned,
-                       float* HWY_RESTRICT out, hwy::ThreadPool& pool) {
-  PROFILER_ZONE("MatVec");
+// Stores dot products of rows with `vec_aligned` + add to a buffer (if kAdd),
```
Comment (jan-wassenberg) [inline ops.h:241]: If we have other changes, consider changing 'add to' to 'add the values from `add`'.

File: ops.h [hunk 2]
```diff
@@ -277,69 +333,105 @@ static HWY_NOINLINE HWY_MAYBE_UNUSED void GeluMulToBF16(
   }
 }
 
+template <class D, HWY_IF_F32_D(D)>
+static HWY_INLINE hn::Vec<D> Sigmoid(D d, hn::Vec<D> v) {
+  using VF = hn::Vec<D>;
+  // Chebyshev polynomial coefficients for rational approximation
+  const VF c0 = Set(d, 0.00949107017368078f);
```
Comment (jan-wassenberg) [inline ops.h:340]: Let's add hn:: to all ops, otherwise it will break on SVE (and some people are indeed targeting that).

File: ops.h [hunk 3]
```diff
@@ -277,69 +333,105 @@ static HWY_NOINLINE HWY_MAYBE_UNUSED void GeluMulToBF16(
...
+  const VF invtwelve = Set(d, 1.0f / 12.0f);
+  const VF lo = Set(d, -11.5f);
+  const VF hi = Set(d, 11.5f);
+
+  VF f = Clamp(v, lo, hi);
+  f = Mul(f, invtwelve);
+  VF f2 = Add(f, f);
+
+  VF a1 = MulAdd(f2, c0, c1);
+  VF a2 = Sub(MulAdd(f2, a1, c2), c0);
```
Comment (jan-wassenberg) [inline ops.h:366]: Is this correct? Looks like we could just pre-bake c2-c0?

File: ops.h [hunk 4]
```diff
@@ -277,69 +333,105 @@ static HWY_NOINLINE HWY_MAYBE_UNUSED void GeluMulToBF16(
...
+  VF a3 = Sub(MulAdd(f2, a2, c3), a1);
+  VF a4 = Sub(MulAdd(f2, a3, c4), a2);
+  VF f0 = Sub(MulAdd(f, a4, c5), a3);
+
+  VF b1 = MulAdd(f2, d0, d1);
+  VF b2 = Sub(MulAdd(f2, b1, d2), d0);
```
Comment (jan-wassenberg) [inline ops.h:372]: Here also.

File: ops.h [hunk 5]
```diff
@@ -277,69 +333,105 @@ static HWY_NOINLINE HWY_MAYBE_UNUSED void GeluMulToBF16(
...
+  VF b3 = Sub(MulAdd(f2, b2, d3), b1);
+  VF f1 = Sub(MulAdd(f, b3, d4), b2);
+
+  return Div(f0, f1);
```
Comment (jan-wassenberg) [inline ops.h:376]: Should we guard against div by zero?

File: ops_test.cc [hunk 6]
```diff
@@ -356,6 +356,155 @@ void TestAllCreateDistribution() {
   TestCreateDistribution<5000>();
 }
 
+template <size_t kOuter, size_t kInner>
+CompressedArray<float, kOuter * kInner> GenerateMat(size_t offset) {
+  hwy::ThreadPool pool(1);
```
Comment (jan-wassenberg) [inline ops_test.cc:361]: Should be 0 (main thread only) or probably a few more (3..4?)

File: ops_test.cc [hunk 7]
```diff
@@ -356,6 +356,155 @@ void TestAllCreateDistribution() {
...
+  }
+  return out;
+}
+
+template <size_t length>
+void AssertClose(const hwy::AlignedFreeUniquePtr<float[]>& a,
+                 const hwy::AlignedFreeUniquePtr<float[]>& b) {
+  for (size_t idx = 0; idx < length; idx++) {
+    const float rel_abs_delta =
+        std::abs(a[idx] - b[idx]) / std::max(a[idx], b[idx]);
```
Comment (jan-wassenberg) [inline ops_test.cc:408]: Also abs of the max? What if both are negative?

**Review by jan-wassenberg — CHANGES_REQUESTED** [review]

File: ops.h (hunk 5 above)
Comment (veluca93) [reply ops.h:377]: No, f1 is >= 0.05 over the real line.

File: ops_test.cc (hunk 7 above)
Comment (veluca93) [reply ops_test.cc]: Thanks for catching that!

**Review by jan-wassenberg — APPROVED** [review]

**Pull request merged**